components/modal/index.jsx

```diff
@@ -181,6 +188,9 @@ class Modal extends React.Component {
 	componentWillMount () {
 		this.generatedId = shortid.generate();
 		checkProps(MODAL, this.props);
+		if (this.props.ariaHideApp) {
+			checkAppElementIsSet();
```

Comment:
If consumer hasn't opted-out, check if the app element was set. The app-launcher component also uses this modal internally so I only need to add this check here.